.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,38 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/rust
+{
+	"name": "Rust for reth",
+
+	"build": {
+		"dockerfile": "./Dockerfile",
+		"context": "."
+	},
+
+	// Use 'mounts' to make the cargo cache persistent in a Docker Volume.
+	// "mounts": [
+	// 	{
+	// 		"source": "devcontainer-cargo-cache-${devcontainerId}",
+	// 		"target": "/usr/local/cargo",
+	// 		"type": "volume"
+	// 	}
+	// ]
+
+	// Features to add to the dev container. More info: https://containers.dev/features.
+	"features": {
+		"ghcr.io/devcontainers/features/common-utils:2": {},
+		"ghcr.io/devcontainers/features/rust:1": {},
+		"ghcr.io/devcontainers/features/git:1.3.0": {}
+	},
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
```

Comment:
should we add default reth ports here?

Reply:
we certainly could - but it all depends on the use case for whether you'd want those forwarded back to your workstation or left exposed on the "remote" machine

it all depends on whether you'd then run anything locally to talk to those forwarded ports or whether you'd run things from inside a VS Code terminal (for example) that runs within the docker container and therefore has access to all ports

for background info - I used a nice big server (112 cores, 512GB RAM, many TBs of SSD) as a development server and then dev from my OSX laptop or Windoze workstation